Exercise_1.java

```diff
@@ -1,44 +1,64 @@
-class Stack { 
-    static final int MAX = 1000; 
-    int top; 
-    int a[] = new int[MAX]; // Maximum size of Stack 
-  
-    boolean isEmpty() 
-    { 
-        //Write your code here 
-    } 
-
-    Stack() 
-    { 
-        //Initialize your constructor 
-    } 
-  
-    boolean push(int x) 
-    { 
-        //Check for stack Overflow
-        //Write your code here
-    } 
-  
-    int pop() 
-    { 
-        //If empty return 0 and print " Stack Underflow"
-        //Write your code here
-    } 
-  
-    int peek() 
-    { 
-        //Write your code here
-    } 
-} 
-  
-// Driver code 
-class Main { 
-    public static void main(String args[]) 
-    { 
-        Stack s = new Stack(); 
-        s.push(10); 
-        s.push(20); 
-        s.push(30); 
-        System.out.println(s.pop() + " Popped from stack"); 
-    } 
+
+class Stack {
+	static final int MAX = 1000;
+	int top = -1;
+	int a[]; // Maximum size of Stack
+
+	boolean isEmpty() {
+		boolean isEmpty = false;
+		if (top == -1)
+			isEmpty = true;
+
+		return isEmpty;
+	}
+
+	Stack() {
+		a = new int[MAX];
+	}
+
+	boolean push(int x) throws Exception {
+		/* Check for stack Overflow */
+		if (top == MAX)
+			throw new Exception("Stack Overflow");
+
+		top++;
+		a[top] = x;

```

Comment:
please check the if condition for overflow, top == Max-1
if we compare with max at 999 it will get overflow for the next push

Reply:
You are right, I updated the file.
Thanks.